portfolio/src/test/java/com/google/sps/ShortestTaskFirstTest.java

```diff
@@ -0,0 +1,171 @@
+
+    Collection<CalendarEvent> events = Arrays.asList();
+    Task task1 = new Task("Task 1", "First task", TIME_30_MINUTES, LOW_PRIORITY);
+    Collection<Task> tasks = Arrays.asList(task1);
+    ScheduledTask scheduledTask1 = new ScheduledTask(task1, TIME_0900);
+
+    Collection<ScheduledTask> actual = algorithm.schedule(events, tasks, TIME_0900, TIME_1700);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask1);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Makes sure that tasks are being scheduled from shortest to longest.
+  @Test
+  public void sortingTasksCorrectly() {
+    ShortestTaskFirst algorithm = new ShortestTaskFirst();
+
+    Collection<CalendarEvent> events = Arrays.asList();
+    Task task1 = new Task("Task 1", "First task", TIME_30_MINUTES, LOW_PRIORITY);
+    Task task2 = new Task("Task 2", "Second task", TIME_30_MINUTES, LOW_PRIORITY);
+    Task task3 = new Task("Task 3", "Third task", TIME_20_MINUTES, LOW_PRIORITY);
+    Task task4 = new Task("Task 4", "Fourth task", TIME_45_MINUTES, LOW_PRIORITY);
+    Collection<Task> tasks = Arrays.asList(task1, task2, task3, task4);
+    // Scheduled tasks throughout the file are meant to correspond with the
+    // same numbered regular task, they are not numbered by their startTime.
+    ScheduledTask scheduledTask1 = new ScheduledTask(task1, TIME_0920);
+    ScheduledTask scheduledTask2 = new ScheduledTask(task2, TIME_0950);
+    ScheduledTask scheduledTask3 = new ScheduledTask(task3, TIME_0900);
+    ScheduledTask scheduledTask4 = new ScheduledTask(task4, TIME_1020);
+
+    Collection<ScheduledTask> actual = algorithm.schedule(events, tasks, TIME_0900, TIME_1700);
+    Collection<ScheduledTask> expected =
+        Arrays.asList(scheduledTask3, scheduledTask1, scheduledTask2, scheduledTask4);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Makes sure that tasks are being scheduled only during periods of time that
+  // contain no events.
+  @Test
+  public void schedulingOnlyInAvailableTimes() {
+    ShortestTaskFirst algorithm = new ShortestTaskFirst();
+    Collection<CalendarEvent> events =
+        Arrays.asList(
+            new CalendarEvent("Event 1", TIME_0920, TIME_1000),
+            new CalendarEvent("Event 2", TIME_1030, TIME_1700));
+
+    Task task1 = new Task("Task 1", "First task", TIME_20_MINUTES, LOW_PRIORITY);
+    Task task2 = new Task("Task 2", "Second task", TIME_30_MINUTES, LOW_PRIORITY);
+    Collection<Task> tasks = Arrays.asList(task1, task2);
+    ScheduledTask scheduledTask1 = new ScheduledTask(task1, TIME_0900);
+    ScheduledTask scheduledTask2 = new ScheduledTask(task2, TIME_1000);
+
+    Collection<ScheduledTask> actual = algorithm.schedule(events, tasks, TIME_0900, TIME_1700);
+    Collection<ScheduledTask> expected = Arrays.asList(scheduledTask1, scheduledTask2);
+    Assert.assertEquals(expected, actual);
+  }
+
+  // Makes sure that no errors are thrown for events outside working hours.
+  @Test
```

Comment:
Maybe we could add another test for some events that lie partially outside the working hours. For example, for a 9-5 working hour, an event could be from 8:30 to 9:30, and the test would make sure that the first scheduled task is not at 9 or something.

Reply:
Done. Broke this test into two new ones, events partially/totally outside working hours.